### PR TITLE
Map 1d arrays to Seq Mems

### PR DESCRIPTION
**Pull request by calebmkim**
Changed Calyx Backend to map 1d arrays to Sequential memories.

### CODE REVIEW

**Comment by rachitnigam** [comment]
Wowow! Quick as usual @calebmkim! The PR looks good but there is one challenge when changing Dahlia code: the Calyx tests will start failing. This is because the Calyx repo contains a bunch of tests that compile Dahlia programs to Calyx. When code generation changes in this repo are merged, Github Actions picks it up, tries to compile Dahlia programs to Calyx and complains when they no longer match.

Long story short, can you update the Dahlia tests in the Calyx repo by running `runt -i 'dahlia|examples`. Easiest way to do it is to merge this, then quickly open a new PR in the Calyx repo and see which tests fail. Then update and merge
